packages/zone.js/lib/node/promise.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+Zone.__load_patch('ZoneAwarePromise', (global: any, Zone: ZoneType, api: _ZonePrivate) => {
+  const __symbol__ = api.symbol;
+  const symbolThen = __symbol__('then');
+
+  api.onUnhandledError = (e: any) => {
+    if (api.showUncaughtError()) {
+      const rejection = e && e.rejection;
+      if (rejection) {
+        console.error(
+            'Unhandled Promise rejection:',
+            rejection instanceof Error ? rejection?.message : rejection,
+            '; Zone:', (<Zone>e.zone).name, '; Task:', e.task && (<Task>e.task).source,
+            '; Value:', rejection, rejection instanceof Error ? rejection.stack : undefined);
+      } else {
+        console.error(e);
+      }
+    }
+  };
+
+  api.microtaskDrainDone = () => {};
+
+  const symbolThenPatched = __symbol__('thenPatched');
+
+  function patchThen(Ctor: Function) {
+    const proto = Ctor.prototype;
+    if ((Ctor as any)[symbolThenPatched] === true) {
+      return;
+    }
+
+    const prop = Object.getOwnPropertyDescriptor(proto, 'then');
+    if (prop && (prop.writable === false || !prop.configurable)) {
+      // check Ctor.prototype.then propertyDescriptor is writable or not
+      // in meteor env, writable is false, we should ignore such case
+      return;
+    }
+
+    const originalThen = proto.then;
+    // Keep a reference to the original method.
+    proto[symbolThen] = originalThen;
+
+    const makeResolver = function(resolveFunc: any, zone: Zone, source: string, isReject: boolean) {
+      if (!resolveFunc) {
+        return resolveFunc;
+      }
+      return (val: any) => {
+        const task = zone.scheduleMicroTask(source, () => {
+          return typeof resolveFunc === 'function' ? resolveFunc(val) :
+              isReject                             ? Promise.reject(val) :
+                                                     val;
```

Comment:
Minor suggestion for readability:
```suggestion
          return typeof resolveFunc === 'function'
              ? resolveFunc(val)
              : (isReject ? Promise.reject(val) : val);
```

Reply:
Got it, fixed, thank you.